kale/embed/mpca.py

```diff
@@ -93,38 +129,57 @@ def fit(self, X):
         return self
 
     def transform(self, X):
-        """
-        Parameter:
-            X: array-like, shape (I1, I2, ..., n_samples)
-        ----------
-        Return:
-            Transformed data, array-like, shape (i1, i2, ..., n_samples)
+        """Perform dimension reduction on X
+
+        Args:
+            X (ndarray): shape (I1, I2, ..., n_samples)
+
+        Returns:
+            ndarray: Transformed data, shape (i1, i2, ..., n_samples)
         """
         n_spl = X.shape[-1]
         for i in range(n_spl):
             X[..., i] = X[..., i] - self.Xmean
-        return multi_mode_dot(X, self.tPs, modes=[m for m in range(self.n_dim - 1)])
+        return multi_mode_dot(X, self.tPs, modes=[m for m in range(self.ndim - 1)])
 
     def inverse_transform(self, X):
-        """
-        Parameter:
-            X: array-like, shape (i1, i2, ..., n_samples)
-        ----------
-        Return:
-            Data in original shape, array-like, shape (I1, I2, ..., n_samples)
-        """
-        return multi_mode_dot(X, self.tPs, modes=[m for m in range(self.n_dim - 1)], transpose=True)
+        """Transform data in the shape of reduced dimension 
+        back to the original shape
 
+        Args:
+            X (ndarray): shape (i1, i2, ..., n_samples), where i1, i2, ... 
```

Comment:
Reduce from I1 to i1 may seem weird, particularly to multilinear algebra notations. I usually use P1, ..Pn, following conventions.

Reply:
Yes, we should make it clearer and reduce barriers here. I followed the scikit-learn documentation here (https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html#sklearn.decomposition.PCA.inverse_transform). But it can be confusing for some users. I will change it later.

Reply:
Good to see you are following top examples. Then maybe that's OK. Of course, sometimes, we can do better if we have a strong justification.